src/services/db/GitFileSystemService.ts

```diff
@@ -479,6 +483,50 @@ export default class GitFileSystemService {
   // TODO: Delete a file
   async delete() {}
 
-  // TODO: Get the latest commit of branch
-  async getLatestCommitOfBranch() {}
+  isGitLocalDiskRawCommitData(
+    commit: unknown
+  ): commit is GitLocalDiskRawCommitData {
+    return (
+      !!commit &&
+      (commit as GitLocalDiskRawCommitData).author_name !== undefined &&
+      (commit as GitLocalDiskRawCommitData).author_email !== undefined &&
+      (commit as GitLocalDiskRawCommitData).date !== undefined &&
+      (commit as GitLocalDiskRawCommitData).message !== undefined &&
+      (commit as GitLocalDiskRawCommitData).hash !== undefined
```

Comment:
Two comments here:

1. is it better to check and make sure all properties of `commit` are `string` according to the definition of `GitLocalDiskRawCommitData`? 
2. nit: `commit` is being cast on multiple lines here, to make it more concise, we can cast it once and reuse the variables for multiple times. 
Suggestions incorporating the above two poitns:
```suggestion
      const c = commit as GitLocalDiskRawCommitData;
      !!commit &&
      typeof c.author_name === 'string' &&
      typeof c.author_email === 'string' &&
      typeof c.date === 'string' &&
      typeof c.message === 'string' &&
      typeof c.hash === 'string'
```

Reply:
updated!